S5T33: Добавить главную страницу с возможностью выбора роли

artbuzzon opened the pull request.

artbuzzon committed:
user selection banner added

artbuzzon committed:
user selection banner added

artbuzzon committed:
cosmetics

artbuzzon committed:
cosmetics

artbuzzon commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Добавить главную страницу с возможностью выбора роли](https://trello.com/c/JpjkIWQp/33-%D0%B4%D0%BE%D0%B1%D0%B0%D0%B2%D0%B8%D1%82%D1%8C-%D0%B3%D0%BB%D0%B0%D0%B2%D0%BD%D1%83%D1%8E-%D1%81%D1%82%D1%80%D0%B0%D0%BD%D0%B8%D1%86%D1%83-%D1%81-%D0%B2%D0%BE%D0%B7%D0%BC%D0%BE%D0%B6%D0%BD%D0%BE%D1%81%D1%82%D1%8C%D1%8E-%D0%B2%D1%8B%D0%B1%D0%BE%D1%80%D0%B0-%D1%80%D0%BE%D0%BB%D0%B8)

bercly0b commented on src/components/StartGameBanner/start-game-banner.tsx:
```diff
@@ -0,0 +1,42 @@
+    const [isOpen, setIsOpen] = useState(false)
+    const [playerRole, setPlayerRole] = useState('')
+
+    const toggleModal = () => {
```
я бы в `useCallback` обернул это

bercly0b commented on src/components/StartGameBanner/start-game-banner.tsx:
```diff
@@ -0,0 +1,42 @@
+        setIsOpen(!isOpen)
+    }
+
+    const setRole = (role: string) => {
```
и это)

alex1056 committed:
Тесты на: input, button, modal (#28)

artbuzzon committed:
T18: Механика игры на Websocket (#25)

* workign version of server done

* first working version

* mark up for game screen done

* auth refactored for id on the server

* chat done

* После pull, router->Router

* S6T25: Добавить Redux-Thunk (Saga) в запросы, S6T31: Добавляет API для изменения профиля пользователя. (#22)

* sass loader added

* Redux хранилище и тестовый компонент для проверки

* Асинхр. запросы с redux-saga

* Redux хранилище и тестовый компонент для проверки

* Компонент Profile с redux-saga

* T11: Экран начала игры (#15)

* nvmrc added

* sass loader added

* Донастройки конфигов + input версия 1

* Оставил только настройки

* waiting for users component added

* cosmetics

* S5T22: Модальное окно (#12)

* modal started

* modal done

* cosmetics

* cosmetics

* S5T23: Add User component (#17)

Co-authored-by: mshcherbak <mshcherbak@future.com>

* S5T21: Компонент Header (навигация) (#14)

* nvmrc added

* sass loader added

* Донастройки конфигов + input версия 1

* Оставил только настройки

* Header

* fix согласно комментариям

Co-authored-by: Mike <mik.arbuzov@gmail.com>

* S5T2: Главная страница, S5T13: Завершение игры (например, уведомление с кнопками «Повторить» или «Вернуться в главное меню»). (#13)

* nvmrc added

* sass loader added

* Донастройки конфигов + input версия 1

* Оставил только настройки

* Добавлен компонент Layout

На главную страницу добавили форму авторизации

* Окончание игры: 2 кнопки - Повторить, Главное меню

* S5T22: Модальное окно (#12)

* modal started

* modal done

* cosmetics

* cosmetics

* S5T23: Add User component (#17)

Co-authored-by: mshcherbak <mshcherbak@future.com>

* S5T21: Компонент Header (навигация) (#14)

* nvmrc added

* sass loader added

* Донастройки конфигов + input версия 1

* Оставил только настройки

* Header

* fix согласно комментариям

Co-authored-by: Mike <mik.arbuzov@gmail.com>

* Fix after review

* sass loader added

* Донастройки конфигов + input версия 1

* Добавлен компонент Layout

На главную страницу добавили форму авторизации

* Окончание игры: 2 кнопки - Повторить, Главное меню

* Fix after review

Co-authored-by: Mike <mik.arbuzov@gmail.com>
Co-authored-by: artbuzzon <62429835+artbuzzon@users.noreply.github.com>
Co-authored-by: Maksim Shcherbak <dragoonz@yandex.ru>
Co-authored-by: mshcherbak <mshcherbak@future.com>

* S5T3: Страница входа (#19)

* nvmrc added

* sass loader added

* Донастройки конфигов + input версия 1

* Оставил только настройки

* Авторизация

Co-authored-by: Mike <mik.arbuzov@gmail.com>

* S5T8: forum & detail forum pages (#16)

Co-authored-by: mshcherbak <mshcherbak@future.com>

* S6T24: Добавить Redux-хранилище (#20)

* sass loader added

* Redux хранилище и тестовый компонент для проверки

* Убрал лишнее

Co-authored-by: Mike <mik.arbuzov@gmail.com>

* S5T5: Страница пользователя (#18)

* Компонент Profile с детьми 100%

* Редактирование профиля убрал в Popup

* Fix согласно ревью

* S5T3: Страница входа (#19)

* nvmrc added

* sass loader added

* Донастройки конфигов + input версия 1

* Оставил только настройки

* Авторизация

Co-authored-by: Mike <mik.arbuzov@gmail.com>

* S5T8: forum & detail forum pages (#16)

Co-authored-by: mshcherbak <mshcherbak@future.com>

* S6T24: Добавить Redux-хранилище (#20)

* sass loader added

* Redux хранилище и тестовый компонент для проверки

* Убрал лишнее

Co-authored-by: Mike <mik.arbuzov@gmail.com>

* Добавил useCallback

* Fixed conflicts

Co-authored-by: Mike <mik.arbuzov@gmail.com>
Co-authored-by: Maksim Shcherbak <dragoonz@yandex.ru>
Co-authored-by: mshcherbak <mshcherbak@future.com>

* Fixed btn style

* waiting for users component added

* cosmetics

Co-authored-by: Alexander <gituser1056@gmail.com>
Co-authored-by: Maksim Shcherbak <dragoonz@yandex.ru>
Co-authored-by: mshcherbak <mshcherbak@future.com>
Co-authored-by: Alexander <64581128+alex1056@users.noreply.github.com>

* Redux хранилище и тестовый компонент для проверки

* Компонент Profile с redux-saga

* Redux хранилище и тестовый компонент для проверки

* Асинхр. запросы с redux-saga

* Fix после ревью

Co-authored-by: Mike <mik.arbuzov@gmail.com>
Co-authored-by: artbuzzon <62429835+artbuzzon@users.noreply.github.com>
Co-authored-by: Maksim Shcherbak <dragoonz@yandex.ru>
Co-authored-by: mshcherbak <mshcherbak@future.com>

* S6T25: Fixed response with null data (#26)

* sass loader added

* Redux хранилище и тестовый компонент для проверки

* Асинхр. запросы с redux-saga

* Redux хранилище и тестовый компонент для проверки

* Компонент Profile с redux-saga

* После pull, router->Router

* Redux хранилище и тестовый компонент для проверки

* Компонент Profile с redux-saga

* Redux хранилище и тестовый компонент для проверки

* Асинхр. запросы с redux-saga

* Fix после ревью

* Fixed with null data response

Co-authored-by: Mike <mik.arbuzov@gmail.com>

* Fixed err in store

* api calls for game moved to a separate file, game server api paths moved to the router

* Тесты на: input, button, modal (#28)

* usecallback added

* workign version of server done

* first working version

* mark up for game screen done

* auth refactored for id on the server

* chat done

* api calls for game moved to a separate file, game server api paths moved to the router

* usecallback added

Co-authored-by: Alexander <gituser1056@gmail.com>
Co-authored-by: Alexander <64581128+alex1056@users.noreply.github.com>
Co-authored-by: Maksim Shcherbak <dragoonz@yandex.ru>
Co-authored-by: mshcherbak <mshcherbak@future.com>

artbuzzon committed:
some usecallbacks added

artbuzzon committed:
some usecallbacks added

artbuzzon committed:
confilicts resolved
